dsl/src/test/scala/org/http4s/dsl/PathInHttpRoutesSpec.scala

```diff
@@ -235,5 +240,23 @@ object PathInHttpRoutesSpec extends Http4sSpec {
           """For input string: "bar""""
         ))
     }
+    "optional flag parameter when present" in {
+      val response =
+        serve(Request(GET, Uri(path = "/flagparam", query = Query.fromString("flag"))))
+      response.status must_== Ok
+      response.as[String] must returnValue("flag present")
+    }
+    "optional flag parameter when present with a value" in {
+      val response =
+        serve(Request(GET, Uri(path = "/flagparam", query = Query.fromString("flag=1"))))
```

Comment:
There's no desire to implement "truthiness" here, correct?  This would still pass if `flag=0`?  I think it should, and does, but just double checking the intent.

Reply:
Yes that was my intent. I was more trying to show that `true` is returned whether there is a value or not - the value is ignored, all that matters is if the parameter key is there or not. I guess I could express that better in this test.

I actually wavered a bit on whether `boolean` was the right return type. WDYT? The alternative was adding a little ADT with `Present` and `Absent` objects, which might be a bit clearer. 

Or ... we could have some arrangement where you get a different type if:

1. The parameter is completely absent
2. The parameter is present but with no value
3. The parameter is present but with a value (and then I guess you'd want the capabilities we have elsewhere for decoding the value)

But IMHO that seemed a bit overcomplicated? I'm not sure if anyone would actually use parameters in this way.

Reply:
`Present` and `Absent` would make clear that it has no opinions on truthiness, but would generally be a bit more inconvenient to use.

I share your uncertainty on the ternary extractor.  It looks good on paper but I'm not sure about on practice.

I don't deal with APIs that use query parameters often, so I don't have a lot of practical experience here.  This is merged, and it made things better.  Let's try it, and if it gives us a rash, we can fix it before 1.0.